elastic/kibana
# [Lens] Shard failure notices make it impossible to use Lens

Closes: #140661

## Summary 

[Lens] Shard failure notices make it impossible to use Lens

## What was done? 

1. Data plugin now exports `ShardFailureOpenModalButton`. 
2. `EuiTextAlign` wrapper was removed from `ShardFailureOpenModalButton` . In Lens we should align button left
3.  `size` param was added for `ShardFailureOpenModalButtonProps`
4. `getRequestMeta` argument was added into `WarningHandlerCallback`
5. Lens: `getTSDBRollupWarningMessages` renamed to `getShardFailuresWarningMessages`
6. Lens: `activeData` now is optional for `onActiveDataChange`


## How to test `Shard Errors` ? 
1. Go to `Dev Tools` and execute 
```
POST a_1/_doc
{
  "d": 100
}
```
7. Go to `Stack Management /Data views` and create a new `Data View` (index pattern should be `a*`)
8. Go to `Lens` and configure new `Visualization` by `a*` index and `d` field
9. Execute the following command in `Dev Tools`. **Important**: should be done in  **new Browser tab** 
```
POST a_2/_doc
{
  "d": "string value"
}
```
10. Switch back to `Lens` and refresh visualization. **Important**: if you reload page, all should be started all from the beginning 🐌 

## Screens

### from editor

<img width="1060" alt="image" src="https://user-images.githubusercontent.com/20072247/195340384-b2056313-9f87-4453-a966-9a1f5a907a5f.png">

### from dashboard

<img width="999" alt="image" src="https://user-images.githubusercontent.com/20072247/195340448-e031dc0e-ff05-4c54-9748-e4b45e48b168.png">


https://user-images.githubusercontent.com/20072247/195375372-0308c6c0-90ca-407b-a597-0f911bb98ee9.mov
